Src/bat_charge.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * bat_charge.c
+ *
+ *  Created on: Sep, 26 2020
+ *      Author: Luke Oxley
+ */
+#include "bat_charge.h"
+
+uint8_t temp_array[READ_MSG_SIZE];
+uint8_t read_byte = 0;
+uint8_t write_data[WRITE_MSG_SIZE + 1];
+
+void init_BQ21062(uint8_t * write_data);
+
+void task_manage_charger() 
+{
+    uint8_t currentF, voltageF, tempF;
+
+    while(1)
+    {
+        read_byte = set_address(STAT0_ADDR, READ_ENABLE);
+        HAL_I2C_Master_Receive(&hi2c1, read_byte, &temp_array[0], READ_MSG_SIZE, 0xFFFF);
+        while (hi2c1.State != HAL_I2C_STATE_READY)
+	    {
+	        //Wait for the send to stop
+	    }
+        currentF = get_bit(temp_array[0], CURRENT_LIM_BIT);
+
+        read_byte = set_address(STAT1_ADDR, READ_ENABLE);
+        HAL_I2C_Master_Receive(&hi2c1, read_byte, &temp_array[0], READ_MSG_SIZE, 0xFFFF);
+        while (hi2c1.State != HAL_I2C_STATE_READY)
+	    {
+	        //Wait for the send to stop
+	    }
+        voltageF = get_bit(temp_array[0], OVERVOLTAGE_BIT);
+        tempF = get_bit(temp_array[0], HOT_BIT);
+
+        //TODO Report faults to something else
+        
+        vTaskDelay(200);
+    }
+
+}
+
+void setChargeEnable(uint8_t enable)
+{
+    write_data[0] = set_address(ICCTRL2_ADDR, WRITE_ENABLE);
+  
+    if(enable) write_data[1] = ICCTRL2_DEF || CHRG_ENABLE;
```

Comment:
don't put conditional expressions on 1 line
should instead be 
if (enable)
{
  write_data[1] = ICCTRL2_DEF | CHRG_ENABLE;
}

also you are using the logical or operator for what appears to be a bitwise operation. The result of this is that the data result will always be 1 if either of the values is > 0
bitwise OR is `|`